lib/kinesis_client/stream/coordinator.ex

```diff
@@ -116,12 +116,21 @@ defmodule KinesisClient.Stream.Coordinator do
 
         {:noreply, %{state | shard_graph: shard_graph, shard_ref_map: map}}
 
-      other ->
-        Logger.info("Stream is not in active state, sleeping... Stream state: #{inspect(other)}")
+      {:error, reason} ->
+        Logger.info("[kcl_ex] error describing stream shards with result #{inspect(reason)}")
+
+        if state.startup_attempt < @max_startup_attempts do
+          :timer.sleep(state.retry_timeout)
```

Comment:
Nit: can we throw some jitter in here to avoid thundering herds, since all of the stream coordinators for all of the streams start at the same time?

Reply:
Good call, will do.